pkg/coscheduling/core/core.go

```diff
@@ -89,10 +98,37 @@ func NewPodGroupManager(client client.Client, snapshotSharedLister framework.Sha
 		podLister:            podInformer.Lister(),
 		permittedPG:          gochache.New(3*time.Second, 3*time.Second),
 		backedOffPG:          gochache.New(10*time.Second, 10*time.Second),
+		podgroups:            map[string]PodGroupInfo{},
 	}
+	podInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		DeleteFunc: func(obj interface{}) {
+			switch t := obj.(type) {
+			case *corev1.Pod:
+				pod := t
+				pgMgr.Unreserve(context.Background(), pod)
```

Comment:
PodDelete event consists of 3 types of events:

- Pod failed
- Pod completed (successfully)
- Pod get deleted

but for completed Pod, we should still count them as part of gang, right? could you also help if integration test covers this case?

Reply:
When pod completed, it will be removed from NodeInfo. CalculateAssignedPods will count pods in NodeInfo, so we did not count completed pods previously.

Reply:
Ok, I will see if integration test covers this case

Reply:
> so we did not count completed pods previously.

True. I'm wondering if we fix this glitch in this PR - in DeleteFunc(), **additionally** check if the Pod is completed, if so, do NOT invalidate it from the `assignedPodsByPG` cache. WDYT?

Reply:
https://github.com/kubernetes-sigs/scheduler-plugins/issues/560#issuecomment-1495316959

We have discussed in this issues about whether we should count completed pods. 
Is there new situation to count completed pods?

Reply:
I see. It seems restart the whole Job is more conventional for now, then let's postpone the idea until new requirement emerges.

Reply:
sure